src/libs/Trie/index.js

```diff
@@ -0,0 +1,115 @@
+import _ from 'underscore';
+import TrieNode from './TrieNode';
+
+class Trie {
+    constructor() {
+        this.root = new TrieNode();
+    }
+
+    /**
+    * Add a word to the Trie
+    * @param {String} word
+    * @param {Object} [metaData] - attach additional data to the word
+    * @param {TrieNode} node
+    * @param {Boolean} [allowEmptyWords] - empty word doesn't have any char, you shouldn't pass a true value for it because we are disallowing adding an empty word
+    * @returns {void}
+    */
+    add(word, metaData, node = this.root, allowEmptyWords = false) {
+        const newNode = node;
+        if (word.length === 0 && !allowEmptyWords) {
+            throw new Error('Cannot insert empty word into Trie');
+        }
+        if (word.length === 0) {
+            newNode.isLeaf = true;
+            newNode.metaData = metaData;
+            return;
+        }
+        if (!newNode.children[word[0]]) {
+            newNode.children[word[0]] = new TrieNode();
+            return this.add(word.substring(1), metaData, newNode.children[word[0]], true);
+        }
+        return this.add(word.substring(1), metaData, newNode.children[word[0]], true);
+    }
+
+    /**
+    * Check if the word is exist in the Trie.
+    * @param {String} word
+    * @returns {Object|null} – the node for the word if it's found, or null if it's not found
+    */
+    search(word) {
+        let newWord = word;
+        let node = this.root;
+        while (newWord.length > 1) {
+            if (!node.children[newWord[0]]) {
+                return null;
+            }
+            node = node.children[newWord[0]];
+            newWord = newWord.substring(1);
+        }
+        return node.children[newWord] && node.children[newWord].isLeaf ? node.children[newWord] : null;
+    }
+
+    /**
+    * Update a word data in the Trie.
+    * @param {String} word
+    * @param {Object} metaData
+    * @returns {void}
```

Comment:
```suggestion
```

Reply:
Makes a lint error.

Reply:
Same as above ... do you have `return null` somewhere where you could instead just use `return`?

Reply:
Done.